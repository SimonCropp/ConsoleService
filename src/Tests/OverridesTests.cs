using System;
using System.Threading;
using System.Threading.Tasks;
using ApprovalTests;
using ConsoleService;
using Xunit;
using Xunit.Abstractions;

public class OverridesTests :
    XunitLoggingBase
{
    [Fact]
    public void Throw_for_onStart()
    {
        var exception = Assert.Throws<Exception>(() => new BadStart());
        Approvals.Verify(exception.Message);
    }

    public class BadStart : ProgramService
    {
        protected override void OnStart(string[] args)
        {
        }

        protected override Task OnStartAsync(string[] args, CancellationToken cancellation)
        {
            return Task.CompletedTask;
        }

        protected override Task OnStopAsync()
        {
            return Task.CompletedTask;
        }
    }

    [Fact]
    public void Throw_for_onStop()
    {
        var exception = Assert.Throws<Exception>(() => new BadStop());
        Approvals.Verify(exception.Message);
    }

    public class BadStop : ProgramService
    {
        protected override void OnStop()
        {
        }

        protected override Task OnStartAsync(string[] args, CancellationToken cancellation)
        {
            return Task.CompletedTask;
        }

        protected override Task OnStopAsync()
        {
            return Task.CompletedTask;
        }
    }

    public OverridesTests(ITestOutputHelper output) :
        base(output)
    {
    }
}