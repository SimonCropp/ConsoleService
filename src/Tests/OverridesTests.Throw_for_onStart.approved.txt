Do not override OnStart or OnStop. Instead place that code in OnStartAsync and OnStopAsync